Dockerfile

```diff
@@ -1,4 +1,28 @@
+FROM mcr.microsoft.com/dotnet/sdk:3.1 as builder
+
+ARG GITHUBSHA OPENWEATHERMAP_APIKEY_RAW 
+
+WORKDIR /prelim-frontend
+
+ENV NODE_OPTIONS="--openssl-legacy-provider"
+
+RUN apt update -y && curl -sL https://deb.nodesource.com/setup_17.x | bash -
+
+RUN apt-get install -y nodejs
+
+COPY . .
+
+RUN sed -i -e "s#{{TAG}}#$GITHUBSHA#g" src/App.fs && sed -i -e "s/|> Program.withConsoleTrace//g" src/App.fs && sed -i -e "s#{{API_KEY}}#$OPENWEATHERMAP_APIKEY_RAW#g" src/App.fs
```

Comment:
This should be split into multiple lines (keep it in a single `RUN` statement) for improved readability.